File: Rust/Dynamic Programming/Jump-Game.rs
struct Solution;

impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut goal = nums.len() - 1;
        for (i, val) in nums.iter().enumerate().rev() {
            if i + *val as usize >= goal {
                goal = i;
            }
        }
        goal == 0
    }
}

fn main() {}

File: Javascript/Dynamic Programming/Jump-Game.js
/**
 *
 * @param {number[]} nums
 * @returns {boolean}
 */
const canJump = (nums) => {
  let goal = nums.length - 1;

  for (let i = nums.length - 1; i >= 0; i--) {
    if (i + nums[i] >= goal) {
      goal = i;
    }
  }

  return goal == 0;
};